WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift

```diff
@@ -13,106 +13,182 @@ struct CommentModerationView: View {
             Divider()
                 .foregroundStyle(Color.DS.Background.secondary)
             VStack(spacing: .DS.Padding.double) {
-                titleHStack
-                mainActionView
-                secondaryActionView
+                switch viewModel.state {
+                case .pending:
+                    Pending(viewModel: viewModel)
+                case .approved(let liked):
+                    Approved(viewModel: viewModel, liked: liked)
+                case .trash, .spam:
+                    TrashSpam(viewModel: viewModel)
+                }
             }
             .padding(.horizontal, .DS.Padding.double)
         }
+        .animation(.smooth, value: viewModel.state)
     }
+}
+
+// MARK: - Subviews
+
+private struct Container<T: View, V: View>: View {
 
-    private var titleHStack: some View {
+    let title: String
+    let icon: V
+    let content: T
+
+    private let animationDuration = 0.25
+
+    private var animation: Animation {
+        return .smooth(duration: animationDuration)
+    }
+
+    init(title: String, @ViewBuilder icon: () -> V = { EmptyView() }, @ViewBuilder content: () -> T) {
+        self.content = content()
+        self.icon = icon()
+        self.title = title
+    }
+
+    var body: some View {
+        VStack(spacing: .DS.Padding.double) {
+            titleHStack
+            content
+        }
+        .padding(.horizontal, .DS.Padding.double)
+        .transition(
+            .asymmetric(
+                insertion: .opacity.animation(animation.delay(animationDuration * 1)),
+                removal: .opacity.animation(animation)
+            )
+        )
+    }
```

Comment:
This `Container` view consolidates common logic used across the state views (Approved, Pending, TrashSpam), including the title and transition logic.

Here's a refined and clear version of your description:

When transitioning between two views, such as from `Pending` to `Approved`, the removal animation for `Pending` starts immediately, while the insertion animation for `Approved` is delayed. This approach helps prevent the **Approve Comment** black button from clashing visually with the  **Reply** view.

Reply:
Would it not be possible to apply this logic without altering the structure as much? I feel like we had to add substantially more code to achieve an improvement on animation (the +- doesn't fully represent it as we removed bunch of #Preview code here). 

I personally found the changes more complicated than the older version.

Reply:
> Would it not be possible to apply this logic without altering the structure as much?

My initial instinct was to enhance the animation without changing the structure too much, but I wasn't able to get the results I wanted. 

For instance, I tried grouping the `titleHStack`, `mainActionView`, and `secondaryActionView` within a `Group` view and then applied the `.transition` to the entire group. It almost worked:

```swift
var body: some View {
    VStack(spacing: .DS.Padding.double) {
        Divider()
            .foregroundStyle(Color.DS.Background.secondary)
        VStack(spacing: .DS.Padding.double) {
            Group {
                titleHStack
                mainActionView
                secondaryActionView
            }
            .transition(
                .asymmetric(
                    insertion: .opacity.animation(animation.delay(animationDuration * 0.8)),
                    removal: .opacity.animation(animation)
                )
            )
        }
        .padding(.horizontal, .DS.Padding.double)
    }
}
```

And here's the result. As you can see, we almost achieved what we were aiming for, but if you look closely, you'll notice that the title doesn’t animate for some reason.

<video src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/5c3797cc-6925-48df-b002-e7cee7258759" ></video>

I've tried another solution of applying the `.transition` modifier on each individual view, but then the code started to get complicated because of the duplicated `.transition` calls.

Reply:
I'm pretty sure there's a way to enhance the animation using the previous structure, but it seemed easier to represent each state as a single unit and then apply the transition to the entire view.

If you feel strongly about keeping the previous structure, I can spend more time to look into this. Let me know what you think. 🙇

Reply:
> I personally found the changes more complicated than the older version.

Could you please elaborate on this a bit more? Although the previous code structure was clear and understandable, I think this approach is also straightforward.

For example, if you look at this code, I think it's pretty clear that each `viewModel.state` is represented by a distinct view. And the implementation of `Pending`, `Approved`,  and `TrashSpam` should be clear too. Let me know if you think otherwise. 🙇 

```swift
    var body: some View {
        VStack(spacing: .DS.Padding.double) {
            Divider()
                .foregroundStyle(Color.DS.Background.secondary)
            VStack(spacing: .DS.Padding.double) {
                switch viewModel.state {
                case .pending:
                    Pending(viewModel: viewModel)
                case .approved(let liked):
                    Approved(viewModel: viewModel, liked: liked)
                case .trash, .spam:
                    TrashSpam(viewModel: viewModel)
                }
            }
            .padding(.horizontal, .DS.Padding.double)
        }
        .animation(.smooth, value: viewModel.state)
    }
```

Reply:
> I personally found the changes more complicated than the older version.

Before there was a View with 3 optional components. Now there's a container view, a Generic View to reduce code duplication, and 3 different views for each state. The call-site is clear, though I think the implementation below is overly complicated imho due to the extra layering, Generic View with dynamic arguments and `onChange` modifier to update the title.

However the animation looks smoother. I'm pretty sure we can achieve this in either implementation but this View has to be resolved for us to proceed with other tickets. I would prefer to merge this than to spend time on the old version's animation. Otherwise we'll stepping on each others' toes.

Please go ahead and merge if the PR is ready and I'll take the changes on my branch.

Reply:
> though I think the implementation below is overly complicated imho due to the extra layering, Generic View with dynamic arguments and `onChange` modifier to update the title.

Thank you for the feedback. I'll create a separate PR to address these points. 🙇